contrib/exporters/core/extend.go

```diff
@@ -0,0 +1,127 @@
+// Structure in yml file is as follows:
+//   transform:
+//     type: <xxx>
+//     extend:
+//       - VAR_NAME1=<gremlin expression with a single substitution string>
+//       - VAR_NAME2=<gremlin expression with a single substitution string>
+type extendGremlin struct {
+	extendStrings []string
+	gremlinClient *client.GremlinQueryHelper
+}
+
+func NewExtendGremlin(cfg *viper.Viper) (interface{}, error) {
+	logging.GetLogger().Infof("entering NewExtendGremlin")
+	gremlinClient := client.NewGremlinQueryHelper(CfgAuthOpts(cfg))
+	extend := cfg.GetStringSlice(CfgRoot + "transform.extend")
+	logging.GetLogger().Infof("NewExtendGremlin, extend = %s", extend)
+        p :=  &extendGremlin{
+		extendStrings: extend,
+		gremlinClient: gremlinClient,
+	}
+	return p, nil
+}
+
+// use the substitution string to extract the value from the provided map
+func (e *extendGremlin) DeriveValue(S map[string]interface{}, varString string) string {
+        substrings := strings.Split(varString, ".")
+        var ii map[string]interface{}
+        var jj map[string]interface{}
+        ii = S
+        for i := 0; i < len(substrings)-1; i++ {
+                str := substrings[i]
+                jj = ii[str].(map[string]interface{})
+                ii = jj
+        }
+	// TODO: verify that each substring is a valid field so we don't get an exception
+        name := ii[substrings[len(substrings)-1]].(string)
+        return name
+}
+
+// use substitution value and perform gremlin query to obtain the desired data enrichment
+func (e *extendGremlin) PerformConversion(origString string, S map[string]interface{}) string {
+	// `%` marks the substitutin string; split the string on `%`
+        substrings := strings.Split(origString, "%")
+        // TODO: verify we have 3 (an odd number of) pieces
+	// convert the value of the substitution string
+        tt := e.DeriveValue(S, substrings[1])
+	// build gremlin expression with the substitution
+        result := substrings[0] + "'" + tt + "'" + substrings[2]
+	// TODO check if result is already in cache; if not, perform gremlin request
+	result2, _ := e.gremlinClient.Query(result)
+	var result3 []string
+	if err := json.Unmarshal(result2, &result3); err != nil {
+		logging.GetLogger().Errorf("PerformConversion: err = %s", err)
+		return ""
+	}
+        return result3[0]
+}
+
+// main function called by pipeline to perform the extension substitutions
+// we expect the data of a single flow per call
+func (e *extendGremlin)  Extend(in interface{}) interface{} {
+	// convert the flow information to a map
+	S1, _ := json.Marshal(&in)
+	var S2 map[string]interface{}
+	S2 = make(map[string]interface{})
+	json.Unmarshal(S1, &S2)
+	// parse the extension string and perform the substitution
+        for _, ee := range e.extendStrings {
+		// TODO do this parsing once at the beginning and save in some data structure
+                substrings := strings.Split(ee, "=")
```

Comment:
you must handling escaping here as the Gremlin expression could contain `=` - for example `g.v().Has("X", "=")`

Reply:
We now split on the first '=', so that any additional '=' is included in the gremlin expression.